Feature/setfiles performance

sknorr opened the pull request:
Could you take a looksie here? This is definitely not yet ready to merge, I just want feedback. Some DAPS functionality is probably broken right now on this branch.

Some other caveats are explained in the commit messages/inline comments.

The goal of this is to improve the performance of DAPS's file list functionality which are somewhat important for my Gedit build integration and for Tomas's vim integration they are too.

The tldr here is that for one specific book (DC-SLES-security), I got this time down from 8.5s to 6.1s on my notebook. Which is a somewhat humble start, I guess.

I have some more ideas on improving that time which I don't know how to realize yet:

1. I noticed that the first xsltproc call takes more than twice as long when called from within make (1.7s v/ 3.6s for DC-SLES-security, as measured with `time`). My (wholly uneducated) guess is that this might have something to do with disabled threading, but I did not see a `-j` parameter or similar anywhere ... Ideas?

2. Checking for profiling is really expensive (at least when running through make, but not so much when running xsltproc directly, interestingly) and we always do it for the whole document, even though we only need to do it for the things that are under `$rootid`. However, as the Makefile is set up right now, we only determine whether we have a rootid after the main XSLT has run. Any good ideas for moving that logic further up?

Commit:
Separate cross.compare into inner and outer template

Commit:
Save a bit of work when checking for profiling in list-src-files

We always checking for a whole bunch of attributes despite the fact that
we generally only use two or three at once. Therefore, it seemed to make
sense to only check for what we needed even though having to use the
local-name() function meant that each individual check was going to be
slower.

One important caveat is currently that the values of profiling
attributes must not include spaces. We surely could find a better
separation character than spaces though.

The other important caveat is that I only checked with list-src-files
and not with anything that needs a full profiling run, so profiling is
probably broken right now.

Crude measurement of list-src-files with DC-SLES-security [1]:
w/o this patch:  8.6s
with this patch: 8.0s
w/o any type of profiling ($profiling.enabled=0): 5.9s (still ouch!)

[1]
rm /tmp/measure; for i in $(seq 1 15); do rm -r build; \
/usr/bin/time -f '%e' $DAPS -d DC-SLES-security list-srcfiles \
> /tmp/list 2>> /tmp/measure; done; \
cat /tmp/mess-o | \
awk '{ sum += $1; n++ } END { if (n > 0) print sum / n; }'

Commit:
Remove well-formed-ness check for setfiles (WIP, coz commented code)

We are running xsltproc anyway and it gets you pretty much the same
errors, so let's save some time. (Unless it is invalid, in which case we
spend some extra seconds.)

Before/after with DC-SLES-security [1]:
before: 8.0s
after:  6.1s

[1] same measurement as in the profiling enhancement commit

tomschr commented on daps-xslt/profiling/check.profiling.xsl:
```diff
@@ -1,169 +1,59 @@
 <xsl:template name="check.profiling">
   <xsl:param name="context" select="."/>
+  <xsl:param name="attributes" select="$profiling.attributes.enabled"/>
+  <xsl:param name="values" select="$profiling.values.merged"/>
```
[Travis](https://travis-ci.org/openSUSE/daps/builds/239516232?utm_source=github_status&utm_medium=notification) says:

```
make[2]: Entering directory `/home/travis/build/openSUSE/daps'
runtime error: file /home/travis/build/openSUSE/daps/daps-xslt/profiling/base-profile.xsl line 56 element call-template
Variable 'profiling.attributes.enabled' has not been declared.
[...]
runtime error: file /home/travis/build/openSUSE/daps/daps-xslt/profiling/base-profile.xsl line 56 element call-template
Variable 'profiling.values.merged' has not been declared.
```

tomschr committed:
Move all profiling to check.profiling.xsl

This is an attempt to collect all profiling stuff into one file.
May or may not work...
